view/src/components/DiffViewWrapper.jsx

```diff
@@ -89,37 +145,60 @@ function getSyntaxHighlightedTokens(hunks) {
  *   amount (LINES_OF_CONTEXT) of lines is shown.
  *
  * For the lines which are not shown, we render buttons that allow user to show these lines.
- * @param {Object} props
+ * @param {Object} props - Note: The old/new props can be null when showing only one version
+ *   of the function.
  * @param {string} props.oldCode - Source code of file containing old version of the function.
+ * @param {string} props.newCode - Source code of file containing new version of the function.
+ *   If oldCode or newCode is empty string, then we are showing only one version of function.
  * @param {string} props.diff - Output of unified diff of old and new source code.
  * @param {number} props.oldStart - Line where the old version of the function starts.
  * @param {number} props.newStart - Line where the new version of the function starts.
  * @param {number} props.oldEnd - Line where the old version of the function ends.
+ * @param {number} props.newEnd - Line where the new version of the function ends.
  * @param {Boolean} [props.showDiff=true] - True to color differences in function.
- * @param {Number} [props.linesToShow=null] - Lines (old, new) of function which
- *                                            should be shown and highlighted.
+ * @param {Number} [props.linesToShow] - Lines (old, new) of function which
+ *   should be shown and highlighted. Can be undefined/null if no line to show.
+ *   Old/new line can be -1 if showing only one version of function.
  */
 export default function DiffViewWrapper({
   oldCode,
+  newCode,
   diff,
   oldStart,
   newStart,
   oldEnd,
+  newEnd,
   showDiff = true,
   linesToShow = null,
 }) {
+  /* Showing code only for one version of a function? */
+  const onlyOneSide = (oldCode === '' || newCode === '');
   /* getting object which Diff component needs */
   const file = useMemo(() => {
-    if (diff !== '') {
-      const [fileObject] = parseDiff(diff);
-      return fileObject;
+    let returnValue;
+    if (onlyOneSide) {
+      const onlyOld = (newCode === '');
+      const code = onlyOld ? oldCode : newCode;
+      const start = onlyOld ? oldStart : newStart;
+      const end = onlyOld ? oldEnd : newEnd;
```

Comment:
Nit: can JS do something like this?
```suggestion
      const [code, start, end] = onlyOld ? [oldCode, oldStart, oldEnd] : [newCode, newStart, newEnd];
```